python/lsst/the/monster/isolate_and_transform.py

```diff
@@ -0,0 +1,149 @@
+import os
+from smatch import Matcher
+
+from lsst.pipe.tasks.isolatedStarAssociation import IsolatedStarAssociationTask
+
+from lsst.the.monster.splinecolorterms import ColortermSpline
+from lsst.the.monster.refcats import GaiaXPInfo, GaiaDR3Info, SkyMapperInfo, PS1Info, VSTInfo
+from lsst.the.monster.utils import read_stars
+
+__all__ = ["MatchAndTransform"]
+
+"""
+This Python script takes a sharded catalog and outputs a shard transformed to
+DES bandpasses.
+
+The following steps are performed:
+1. Read in the Gaia shard.
+2. Run the IsolatedStarAssociationTask on the Gaia shard.
+3. Read in each catalog.
+4. Match each catalog with the Gaia DR3 catalog.
+5. For each band, interpolate the DES magnitudes.
+6. Save the catalog shard.
+"""
+
+
+class MatchAndTransform:
+    """Match catalogs to Gaia and transform them to 'the_monster'
+       reference frame.
+
+    Parameters
+    ----------
```

Comment:
This docstring is in the location for `__init__`, not `run()`.  So should be moved below.

Furthermore, I think that the class could take in `GaiaDR3Info` and `testing_mode` and set those there rather than being class vars (though for this code it doesn't matter too much)

Reply:
In fact, I think all of everything here should be in the `__init__` and not the `run` except for htmid.

Then you can, in the `__init__`, instantiate all the `info` objects.